src/items/extern-crates.md

```diff
@@ -68,16 +68,24 @@ need for specifying `extern crate`.
 
 <!--
 Possible upcoming changes that will change this:
-
 The `extern_crate_item_prelude` unstable feature allows `extern crate` to
 update the extern prelude in certain situations, see
 https://github.com/rust-lang/rust/pull/54658
-
 Unstable `--extern proc_macro` flag that would force the crate into the
 extern prelude.
     https://github.com/rust-lang/rust/pull/54116
 -->
 
+## Underscore Imports
+
+An external crate dependency can be declared without binding its name in scope
+by using an underscore with the form `extern crate foo as _`. This may be
+useful for crates that only need to be linked, but are never referenced, and
+will avoid being reported as unused.
+
+The `#[macro_use]` attribute will work as usual and import the macro names
```

Comment:
Is there somewhere to hyperlink "as usual" to?

Reply:
Not really. I added a link to where `macro_use` is described, but it is very brief.  I actually just today learned about the macro-use prelude, and I'm not sure if it is wise to mention it here since it is not described anywhere.  

In general I think it would be amazing to have good documentation on name resolution — something that explained the different namespaces, the different preludes, the precedence and order, macro resolution, maybe ribs?, etc.  I added docs on "extern prelude", but I don't really understand these things.  I think if I did understand them, my mental model of how Rust works would be substantially improved.  I'm not even sure where to start learning it (is everything in librustc_resolve? and macros are mostly in libsyntax?). The current [rustc-guide docs](https://rust-lang.github.io/rustc-guide/name-resolution.html) are kinda incomplete.

Reply:
You should have a Q&A session with @petrochenkov... =P